functorch/functorch/csrc/BatchRulesFactory.cpp

```diff
@@ -9,6 +9,26 @@
 
 namespace at { namespace functorch {
 
+template <typename A, A a, typename C>
+struct NewBlahBatchRuleHelperSymInt;
+
+template <typename F, F Func, typename A, typename B, typename... T>
+struct NewBlahBatchRuleHelperSymInt<F, Func, typelist<A, B, T...>> {
+  static std::tuple<Tensor,optional<int64_t>> apply(
+      const Tensor& tensor,
+      optional<int64_t> batch_dim,
+      SymIntArrayRef shape,
+      T... extra_args) {
+    const auto bdim_size = tensor.sym_size(batch_dim.value());
+    // TODO: optimize this
+    std::vector<c10::SymInt> new_shape;
+    new_shape.reserve(shape.size() + 1);
+    new_shape.emplace_back(bdim_size);
+    new_shape.insert(new_shape.end(), shape.begin(), shape.end());
+    return std::make_tuple(Func(tensor, new_shape, std::forward<T>(extra_args)...), 0);
+  }
+};
```

Comment:
The code duplication here is irritating (see template below), but I didn't want to mess around with the template too much. cc @zou3519

Reply:
Yeah this is fine, these templates are delicate